test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -111,6 +111,49 @@ session_simulator::rollback_transaction(const std::string &config)
     return (0);
 }
 
+int
+session_simulator::prepare_transaction(const std::string &config)
+{
+    /* 'prepare_transaction' only permitted to be called once in a running transaction. */
+    if (_prepared_txn)
+        WT_SIM_RET_MSG(EINVAL,
+          "'prepare_transaction' only permitted to be called once in a running transaction");
+
+    /* Make sure that the transaction from this session is running. */
+    if (!_txn_running)
+        WT_SIM_RET_MSG(EINVAL, "'prepare_transaction' only permitted in a running transaction");
+
+    timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
+    std::map<std::string, std::string> config_map;
+
+    const std::vector<std::string> supported_ops = {"prepare_timestamp"};
+    const std::vector<std::string> unsupported_ops;
+
+    WT_SIM_RET_MSG(ts_manager->parse_config(config, supported_ops, unsupported_ops, config_map),
+      "Incorrect config (" + config + ") passed in prepare_transaction");
+
+    auto pos = config_map.find("prepare_timestamp");
+    if (pos != config_map.end()) {
+        WT_SIM_RET(ts_manager->validate_hex_value(pos->second, "prepare timestamp"));
+        uint64_t prepare_ts = ts_manager->hex_to_decimal(pos->second);
+        set_prepare_timestamp(prepare_ts);
+    }
+
+    /* A prepared timestamp should have been set at this point. */
+    if (!has_prepare_timestamp())
+        WT_SIM_RET_MSG(EINVAL, "'prepare_transaction' - prepare timestamp is not set");
+
+    /* commit timestamp must not be set before transaction is prepared. */
+    if (_has_commit_ts)
```

Comment:
could we move this check to line 125, that way the validity of the transaction is verified before processing any config?

Reply:
The code flow is similar to wiredtiger such that we get the same errors. Let me know what you think.

Reply:
I will merge this ticket, if this remains a concern to you then we can look this in the next PR for WT-9232